### DISCUSSION TITLE
How to slide out closed PRs / deleted branches

### DISCUSSION THREAD

**Discussion by tir38** [VirtusLab/git-machete#1251] [Q&A] [answered]
After merging/closing a PR and deleting the branch on Github, how can I get machete to slide out that branch? I'd ideally like to do this during a traverse.

**Comment by PawelLipski** [comment, marked as answer]
Short answer: **`git machete slide-out --removed-from-remote`**. This requires your local mirror (`.git/refs/remotes/origin/...`) of remote branch to be removed &mdash; as done by `git fetch --prune`. `git machete traverse --fetch` always runs `git fetch` in `--prune` mode, and IIRC IntelliJ also uses that mode of fetch by default.

Still, git machete should _typically_ detect that you've merged your branch, even if it was a squash &mdash; by looking up a commit with identical contents (_tree_, in gitspeak) in parent branch history. This is under improvement currently by @gjulianm on #1238, to detect even more non-obvious squash-merges ;) This merged-ness of a branch is marked as gray edge in `status`, and `traverse` will suggest to slide-out this branch (instead of rebasing/pushing etc.).

Also, speaking of `traverse`, there's a plan to integrate the behavior of `slide-out --removed-from-remote` into `traverse` as well. So sliding out will not only be suggested for merged branches, but also for the ones removed from remote (#1242). Still, these two categories overlap a lot in practice.
